linguicator_predictor/service.py

```diff
@@ -0,0 +1,158 @@
+import json
+import logging
+import linguicator_predictor
+
+
+async def handle_request(websocket, user_text, request_data):
+    """
+    I handle a request passing the received data to the proper handler.
+
+    :param websocket: websocket object representing the connection
+    :type  websocket: websockets.server.WebSocketServerProtocol
+
+    :param user_text: text the user typed so far
+    :type  user_text: str
+
+    :param request_data: the received request data
+    :type  request_data: str
+
+    :returns: the updated user text
+    :rtype: str
+    """
+    # define request handlers
+    request_handlers = {
+        '/predict': handle_prediction_request,
+        '/feed': handle_feed_request
+    }
+
+    # convert request json to dictionary
+    request_dict = json.loads(request_data)
+
+    # handle the request and return the updated user data
+    return await request_handlers[request_dict['resource']](
+        websocket,
+        user_text,
+        request_dict['id'],
+        request_dict['data']
+    )
+
+
+async def handle_prediction_request(websocket,
+                                    user_text,
+                                    request_id,
+                                    request_data):
+    """
+    :param websocket: websocket object representing the connection
+    :type  websocket: websockets.server.WebSocketServerProtocol
+
+    :param user_text: text the user typed so far
+    :type  user_text: str
+
+    :param request_id: the id of the request being handled
+    :type  request_id: int
+
+    :param request_data: this was added to make all request handlers receive
+                         the same amount of arguments, but it is not necessary
+                         for the function to
+                         work properly
+    :type  request_data: any
+
+    :returns: the unaltered user text
+    :rtype: str
+    """
+    prediction = predict(user_text)
+    await websocket.send(json.dumps({
+        'id': request_id,
+        'code': 200,
+        'data': prediction
+    }))
+    return user_text
+
+
+def predict(user_text):
+    """
+    I make a prediction of how a text could continue.
+
+    :param user_text: the text I will make the prediction on
+    :type  user_text: str
+
+    :returns: the predicted text
+    :rtype: str
+    """
+    prediction = linguicator_predictor.model.predict(user_text,
+                                                     len(user_text) + 30)
+
+    # remove the original text from the prediction
+    prediction = prediction[len(user_text):]
+
+    # remove everything after last ' ' and return
+    return ' '.join(prediction.split()[:-1])
+
+
+async def handle_feed_request(websocket, user_text, request_id, new_text):
```

Comment:
I was thinking about what would make more sense:

a) the front-end define what should be used as an input for the prediction

or 

b) the backend keep track of the current state of each client text connected to it

And I think that maybe `a` is more efficient and more configurable. If you agree we might refactor this MR and remove almost all this method functionality. 

Maybe we should discuss the design of the protocol between front and backstage a little bit more...